src/main/kotlin/g0301_0400/s0382_linked_list_random_node/Solution.kt

```diff
@@ -0,0 +1,40 @@
+package g0301_0400.s0382_linked_list_random_node
+
+// #Medium #Math #Linked_List #Randomized #Reservoir_Sampling
+// #2022_11_24_Time_479_ms_(60.00%)_Space_53.4_MB_(40.00%)
+
+import com_github_leetcode.ListNode
+import java.util.Random
+
+/*
+ * Example:
+ * var li = ListNode(5)
+ * var v = li.`val`
+ * Definition for singly-linked list.
+ * class ListNode(var `val`: Int) {
+ *     var next: ListNode? = null
+ * }
+ */
+@Suppress("kotlin:S2245")
+class Solution(private val head: ListNode?) {
```

Comment:
Please reuse java solution

https://github.com/javadev/LeetCode-in-Java/blob/main/src.save/main/java/g0301_0400/s0382_linked_list_random_node/Solution.java